source/Engine.cpp

```diff
@@ -2265,7 +2261,10 @@ void Engine::DoCollisions(Projectile &projectile)
 			// "safe" weapon.
 			Point hitPos = projectile.Position() + range * projectile.Velocity();
 			bool isSafe = weapon.IsSafe();
-			for(Body *body : shipCollisions.Circle(hitPos, blastRadius))
+			vector<Body *> blastCollisions;
+			blastCollisions.reserve(ships.size());
```

Comment:
```suggestion
			vector<Body *> blastCollisions(ships.size());
```

Reply:
Doesn't that fill the vector with default-constructed elements?

Reply:
ah true forgot about that